### PR TITLE
Add support for media dir environment variable.

### LINKED ISSUE #4: Feature Request: Do not store media in local project directory
First off, this is a very cool project. Thank you for it!

One thing I don't like is that git-sim stores it's media files in the local project directory, and it's easy to accidentally commit them to the git repo if it hasn't been added to the .gitignore file. Most of the time I think the images/videos are only needed temporarily, and could be deleted. There is the `--media-dir` parameter, but having to provide it every time you use git-sim is cumbersome.

There's a few different ways you could tackle this problem. Some thoughts I had were:

1. Allow the user to configure a global media directory that would be used instead of default local directory. The user can manually delete the global directory from time to time to clean up the media files.
2. Save the media to the user's temporary directory, where it will eventually get cleaned up automatically. I'm not sure if this is a cross-platform concept or not and if Python has an easy way of getting the temp path, but hopefully.

Maybe you (or others) have other suggestions?

### PR DESCRIPTION
**Pull request by abhijitnathwani**
@initialcommit-io  as discussed, I have implemented the changes for setting the environment variable. Let me know if it requires any other updates. I have tested on Windows/Linux. I don't have access to a Macintosh to test it. Can you please verify this once ?
Fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -339,6 +339,14 @@ Customize the output image/video directory location:
 $ git-sim --media-dir=path/to/output status
 ```
 
+Optionally, set the environment variable `git_sim_media_dir` to default that as the media directory, if no `--media-dir` is provided.
```
Comment (initialcommit-io) [inline README.md:342]: Let's update to something like:

Optionally, set the environment variable `git_sim_media_dir` to set a global default media directory, to be used if no `--media-dir` is provided. Simulated output images/videos will be placed in this location, in subfolders named with the corresponding repo's name.

File: README.md [hunk 2]
```diff
@@ -339,6 +339,14 @@ Customize the output image/video directory location:
...
+
+```console
+$ export git_sim_media_dir=path/to/media/directory
+$ git-sim status
+```
+Note: `--media-dir` takes precedence over the environment variable. If you set the environment and still provide the argument, you'll find the media in the path provided by `--media-dir`
```
Comment (initialcommit-io) [inline README.md:348]: Add a period at the end of the second sentence.

File: git_sim/__main__.py [hunk 3]
```diff
@@ -76,9 +77,18 @@ def main():
...
+    if sys.platform == 'linux' or sys.platform == 'darwin':
+        repo_name = git.Repo(search_parent_directories=True).working_tree_dir.split('/')[-1]
+    elif sys.platform == 'win32':
+        repo_name = git.Repo(search_parent_directories=True).working_tree_dir.split('\\')[-1]
+
     config.media_dir = os.path.join(os.path.expanduser(args.media_dir), "git-sim_media")
     config.verbosity = "ERROR"
 
+    # If the env variable is set and no argument provideed, use the env variable value
+    if os.getenv('git_sim_media_dir') is not None and args.media_dir == '.':
```
Comment (initialcommit-io) [inline git_sim/__main__.py:89]: I think we can shorten the `if` statement if we remove the `is not None` and just use `if os.getenv('git_sim_media_dir') and args.media_dir == '.':`

File: git_sim/__main__.py [hunk 4]
```diff
@@ -76,9 +77,18 @@ def main():
...
 
+    if sys.platform == 'linux' or sys.platform == 'darwin':
+        repo_name = git.Repo(search_parent_directories=True).working_tree_dir.split('/')[-1]
+    elif sys.platform == 'win32':
+        repo_name = git.Repo(search_parent_directories=True).working_tree_dir.split('\\')[-1]
+
     config.media_dir = os.path.join(os.path.expanduser(args.media_dir), "git-sim_media")
     config.verbosity = "ERROR"
 
+    # If the env variable is set and no argument provideed, use the env variable value
```
Comment (initialcommit-io) [inline git_sim/__main__.py:88]: Typo - "provideed" instead of "provided"

**Review by initialcommit-io — CHANGES_REQUESTED** [review]
Looks like there are multiple similar commits here in this pull request, can you squash it into just 1? And make the changes I requested in the my comments?

**Comment by abhijitnathwani** [comment]
Thanks for the review! I have updated it as per your comments.

**Pull request merged**